source/common/upstream/subset_lb.cc

```diff
@@ -110,6 +112,63 @@ void SubsetLoadBalancer::refreshSubsets(uint32_t priority) {
   update(priority, host_sets[priority]->hosts(), {});
 }
 
+void SubsetLoadBalancer::initSubsetSelectorMap() {
+  selectors_ = std::make_shared<SubsetSelectorMap>();
```

Comment:
Since selectors_ is always initialized, I suggest moving its initialization to the constructor initializer list or better yet initialize it right in the header at the point of definition.

Reply:
@yanavlasov cannot init selectors_ in the header, as it depends on field `subset_selectors_` being initialised in constructor. Moved out this initialisation into separate method, as constructor body is already big and it could be hard to read. What do you think?

Reply:
I was suggesting to move just the first line where the pointer is initialized to point to an empty SubsetSelectorMap object to the header. However the way you have is fine as well.